runtime % support for FreeRTOS V11

malsyned opened the pull request:
In V11, FreeRTOS changed `ulTotalRunTime` from an integer to an array of integers. On SMP, the array has one entry per CPU, but it's still a one-element array even without SMP support compiled in.

Try finding the length of the array with `sizeof(a)/sizeof(a[0])`. If that succeeds, sum all CPU runtimes to get the total runtime. If it fails, assume V10 or earlier and fall back to the previous technique.

malsyned commented:
@hwmland @haneefdm @PhilippHaefele I would love your input on this.

hwmland commented:
Looks OK for me.

Jonarw commented:
Any chance of this getting merged and released soon? I would love to use runtime stats in my FreeRTOS v11 based project!

haneefdm commented:
I tried to merge this but it has conflicts with recently merged PR#45. Sorry for the late merge.

@Jonarw, Could you resolve them please?

malsyned commented:
I'll look into this today, thanks for the heads up

malsyned commented:
OK, I rebased onto the head of `main`. While I was in there, I also reworked this change into a much simpler one that is more consistent with the way the rest of the code is written.

I tested all the latest changes with versions of FreeRTOS that used both an array and a scalar for `ulTotalRunTime`.

Hopefully you don't mind that I rewrote `updateTotalRuntime()` to be an async function. It made the logic a lot simpler to express without changing its interface.
